### DISCUSSION TITLE
Support for apiserver /proxy endpoint?

### DISCUSSION THREAD

**Discussion by stevemathieu** [fabric8io/kubernetes-client#5032] [Q&A]
Hi everyone,

Does the kubernetes-client support the apiserver /proxy endpoint as described in: 
https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster-services/#manually-constructing-apiserver-proxy-urls

I would like to be able to proxy both http and https requests to pods and services running inside of a namespace.

Thank you!

**Comment by shawkins** [comment]
> Does the kubernetes-client support the apiserver /proxy endpoint as described in

It presently does not.  Do you have a reference for how other clients utilize that?

**Reply by stevemathieu** [reply]
Thank you for the response.

A quick search shows most of the official kubernetes clients support the proxy spec.
**Python:**
https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#connect_get_namespaced_service_proxy_with_path

**Ruby:**
https://github.com/kubernetes-client/ruby/blob/master/kubernetes/docs/CoreV1Api.md#connect_get_namespaced_service_proxy_with_path

**Java:**
https://javadoc.io/static/io.kubernetes/client-java-api/18.0.0/io/kubernetes/client/openapi/apis/CoreV1Api.html#connectGetNamespacedServiceProxyWithPath(java.lang.String,java.lang.String,java.lang.String,java.lang.String)

Also, some of the other community maintained client libraries support it.
**Godaddy / Node.JS:**
https://github.com/godaddy/kubernetes-client/blob/master/docs/1.13/ServiceProxyOptions.md#proxy

**Comment by shawkins** [comment]
I've opened a pr for simple Client.raw methods: #5064 

Can you open an issue for dsl proxy method support?  It could be as simple as ServiceResource.proxy(boolean https, String pathAndParameters, String method, String payload) - that would be accessible via client.services.inNamespace("ns").withName("name").proxy(false ...) or it could have a full dsl of its own .proxy(path).https().get()
